### PR TITLE
fix(migration): Fixes mute_till migration

### PR DESCRIPTION
**Pull request by borismelnik**
Fix the problem with `mute_till` migration file:

The PR with `muting community with intervals` was merged with new migration file with an old timestamp which introduced the bug, when user can't login because of `mute_till` column not exists. But no error occurs when creating a new user account.
To cover all cases, this PR introduce the migration file which should recreate `communities_communities` table and copy all data.

Important changes:
- [x] Something worth noting for reviewers.

Closes #

### CODE REVIEW

**Review by jrainville — COMMENTED** [review]
Would `ALTER TABLE communities ADD COLUMN muted_till TIMESTAMP;` be enough

**Comment by Samyoul** [comment]
@borismelnik would you be able to detail a little more in the PR description what the problem is and how this PR fixes that problem?

**Comment by borismelnik** [comment]
> Would `ALTER TABLE communities ADD COLUMN muted_till TIMESTAMP;` be enough

After discussion with @cammellos I realized its not enough :(

**Review by osmaczko — COMMENTED** [review]
I believe it would be enough to just change the migration timestamp and run `make generate`.

```diff
diff --git a/protocol/migrations/sqlite/1685383829_add_communities_mute_till.up.sql b/protocol/migrations/sqlite/1685964184_add_communities_mute_till.up.sql
similarity index 100%
rename from protocol/migrations/sqlite/1685383829_add_communities_mute_till.up.sql
rename to protocol/migrations/sqlite/1685964184_add_communities_mute_till.up.sql
```

**Comment by osmaczko** [comment]
> > Would `ALTER TABLE communities ADD COLUMN muted_till TIMESTAMP;` be enough
> 
> After discussion with @cammellos I realized its not enough :(

Can you please elaborate on that?

**Comment by cammellos** [comment]
> > > Would `ALTER TABLE communities ADD COLUMN muted_till TIMESTAMP;` be enough
> > 
> > 
> > After discussion with @cammellos I realized its not enough :(
> 
> Can you please elaborate on that?

It works for those accounts who didn't run the migration, but if your account did run it (a new account), that would result in an error, as it would try to run the same migration twice.

**Comment by cammellos** [comment]
also, if you were right on that migration (which is very unlikely, since it was out of order), you would get "file not found", as the migration system looks for the latest migration, though in this case I don't think it would happen.

The safest way is always to add a migration in general, renaming the timestamp is less bullet proof, though it might work in some instances (in this case I don't think it would, since we can't make `ALTER TABLE` not fail if the db had already been migrated, if it's was a create table, you can use IF EXISTS)

**Comment by borismelnik** [comment]
@Samyoul @cammellos I can't merge it without approves :D

**Review by jrainville — APPROVED** [review]

**Comment by osmaczko** [comment]
> I believe it would be enough to just change the migration timestamp and run `make generate`.
> 
> ```diff
> diff --git a/protocol/migrations/sqlite/1685383829_add_communities_mute_till.up.sql b/protocol/migrations/sqlite/1685964184_add_communities_mute_till.up.sql
> similarity index 100%
> rename from protocol/migrations/sqlite/1685383829_add_communities_mute_till.up.sql
> rename to protocol/migrations/sqlite/1685964184_add_communities_mute_till.up.sql
> ```

In case anyone still wondering why it wouldn't work well.

image

File: protocol/migrations/migrations.go
Comment (osmaczko) [inline protocol/migrations/migrations.go:1]: Only this file should be re-generated, could you please remove others from the commit?

Comment (borismelnik) [reply protocol/migrations/migrations.go:1]: You mean after rebase? but I got a conflicts there

**Comment by borismelnik** [comment]
> > I believe it would be enough to just change the migration timestamp and run `make generate`.
> > ```diff
> > diff --git a/protocol/migrations/sqlite/1685383829_add_communities_mute_till.up.sql b/protocol/migrations/sqlite/1685964184_add_communities_mute_till.up.sql
> > similarity index 100%
> > rename from protocol/migrations/sqlite/1685383829_add_communities_mute_till.up.sql
> > rename to protocol/migrations/sqlite/1685964184_add_communities_mute_till.up.sql
> > ```
> 
> In case anyone still wondering why it wouldn't work well.
> 
> image

I actually agree, we still not released

File: protocol/migrations/migrations.go
Comment (osmaczko) [reply protocol/migrations/migrations.go:1]: I mean the other `bindata.go` and `migrations.go` files update are not needed.

**Review by cammellos — APPROVED** [review]

**Review by jrainville — APPROVED** [review]

**Pull request merged**